galasa-parent/dev.galasa.framework.api.ras/src/test/java/dev/galasa/framework/api/ras/internal/mocks/MockResultArchiveStoreDirectoryService.java

```diff
@@ -32,8 +35,43 @@ public boolean isLocal() {
 
 
 	@Override
-	public @NotNull List<IRunResult> getRuns(@NotNull IRasSearchCriteria... searchCriteria)
-			throws ResultArchiveStoreException {
+	public @NotNull List<IRunResult> getRuns(@NotNull IRasSearchCriteria... searchCriterias) throws ResultArchiveStoreException {
+		for(IRasSearchCriteria searchCriteria : searchCriterias) {
+			if (searchCriteria instanceof RasSearchCriteriaQueuedFrom) {
+				List<IRunResult> returnRuns = new ArrayList<IRunResult>() ;
+				RasSearchCriteriaQueuedFrom sFrom = (RasSearchCriteriaQueuedFrom) searchCriteria;
+				for (IRunResult run : this.getRunsResults){
+					Boolean compareInstant = sFrom.criteriaMatched(run.getTestStructure());
+					if (compareInstant){
+						returnRuns.add(run);
+					}
+				}
+				this.getRunsResults = returnRuns;
+			} else if (searchCriteria instanceof RasSearchCriteriaRunName) {
+				List<IRunResult> returnRuns = new ArrayList<IRunResult>() ;
+				RasSearchCriteriaRunName sRunName = (RasSearchCriteriaRunName) searchCriteria;
+				for (IRunResult run : this.getRunsResults){
+					Boolean compareInstant = sRunName.criteriaMatched(run.getTestStructure());
+					
+					if (compareInstant){
+						returnRuns.add(run);
+					}
+				}
+				this.getRunsResults = returnRuns;
+			} else if (searchCriteria instanceof RasSearchCriteriaQueuedTo) {
+				List<IRunResult> returnRuns = new ArrayList<IRunResult>() ;
+				RasSearchCriteriaQueuedTo sTo = (RasSearchCriteriaQueuedTo) searchCriteria;
+				for (IRunResult run : this.getRunsResults){
```

Comment:
This block of code is repeated 3 times. Seems that once you have the search criteria, you could move the following to teh end of the function before returning ? 
```
for (IRunResult run : this.getRunsResults){
					Boolean compareInstant = sRunName.criteriaMatched(run.getTestStructure());

					if (compareInstant){
						returnRuns.add(run);
					}
				}
				this.getRunsResults = returnRuns;
```
Otherwise you could put that into a method of it's own and call it 3 times... 

I prefer to only return at the end of methods normally... partly so that kind of refactoring is easier.

Reply:
Why are we implementing sort code in a mock ? Surely the test data we feed in should be sorted as we expect it to be ?

Reply:
minimising duplicate code even in test code is a good idea.